extensions/2.0/Khronos/KHR_texture_procedurals/README.md

```diff
@@ -0,0 +1,1397 @@
+
+4. **Validity**: Ensure that procedurals graphs can be validated both against a schema as well as against reference rendering implementations.
+
+5. **Flexibility**: Support for procedural graphs is additive and can be used in conjunction with traditional texture maps. This allows for a hybrid approach where some textures can be baked and others are generated at runtime or both can be specified.
+
+| Original | Baked |
+| :---: | :---: |
+| <img src="./figures/procedural_marble.png" width=100%> | <img src="./figures/baked_marble.png" width=100%> |
+
+<super>Example of a procedural versus a baked 3D marble texture. In this case re-baking is required for each geometry bound or / and when geometry is transformed</super>
+
+### Definitions
+
+The following is a set of definitions using MaterialX nomenclature to provide context for the procedural graph representation.
+
+* A **Node** is a function that generates or operates upon spatially-varying data.  The MaterialX specification provides the set of standard nodes with precise definitions and also supports the creation of custom nodes.
+
+* **Node Input and Output Ports** The interface for a node’s incoming data is declared through **input ports**, which may be spatially-varying or uniform. The interface for a node’s outgoing data is declared through **output ports**.  
+
+* There is a specific set of supported **Data Types**. Every port must have a data type.
+
+* A **Pattern** is a node that generates or processes simple scalar, vector, and color data, and has access to local properties of any geometry that has been bound to a given material.
+
+* A **Node Graph** is a directed acyclic graph (DAG) of nodes, which may be used to define arbitrarily complex generation or processing networks.  Node Graphs describe a network of "pattern" nodes flowing into shader inputs, or define a complex or layered node in terms of simpler nodes. The former is called a **compound nodegraph** and the latter a **functional nodegraph**.
+
+* Node Graph:
+  * **Inputs** are nodes that define the interface for a graph’s incoming data.
+  * **Outputs** are nodes that define the interface for a graph’s outgoing data.
+
+* A **Node Definition** is a specification for a unique function with a given set of inputs and outputs. The logic for each function is represented either as a `functional graph` or as code. 
+
+## Extension Declaration
+
+Usage is indicated by adding the `KHR_texture_procedurals` extension identifier to the `extensionsUsed` array.
+
+```json
+{
+    "extensionsUsed": [
+        "KHR_texture_procedurals"
+    ]
+}
+``` 
+with the extension object defined in the `extensions array:
+
+```json
+{
+    "extensions": {
+        "KHR_texture_procedurals": {
+            "mimetype": "application/mtlx+json;version=<MaterialX_version>",
+            "procedurals": []
+        }
+    }
+}
```

Comment:
Following up from last call, proposed change is to simply mention
```
"extensions": {
        "KHR_texture_procedurals"
    }
```
Which indicates that gltf texture procedural "structure" is used. 
When gltf texture procedural uses MaterialX 1.39 nodes the extension will be 
```
"extensions": 
                "KHR_texture_procedurals_MaterialX_1.39": {
}
```
Example usage:
```
"extensionsRequired" : [
    "KHR_texture_procedurals"
]
"extensionsUsed" : [
    "KHR_texture_procedurals_MaterialX_1.39"
]

```

Reply:
use case #1 gltf using Mtlx 1.39
extensionsRequired" : [  
    "KHR_texture_procedurals"  
    "KHR_texture_procedurals_MaterialX_1.39"  
]  
use case #2 gltf using Mtlx 2.0
extensionsRequired" : [  
    "KHR_texture_procedurals"  
    "KHR_texture_procedurals_MaterialX_2.00"
]  
use case #3
extensionsRequired" : [  
    "KHR_texture_procedurals"  
    "NV_MDL_texture_procedurals"  
]  
(mixing of procedurals is not allowed, only one node library can be used at a time)
extensionsRequired" : [  
    "KHR_texture_procedurals"  
    "KHR_texture_procedurals_MaterialX_1.39"  
    "KHR_texture_procedurals_MaterialX_2.00"      
]

Reply:
What does it mean to include `NV_MDL_texture_procedurals` is there some co-dependency implied here or just an example ?

Reply:
What does the `extenions` portion look like. Is it still this ?
```json
{
    "extensions": {
        "KHR_texture_procedurals": {
            "procedurals": []
        }
    }
}
```
Or do you need something like this ?
```json
{
    "extensions": {
        "KHR_texture_procedurals_MaterialX_1.39": {
            "procedurals": []
        }
    }
}
```